packages/pyright-scip/src/lib.ts

```diff
@@ -40,19 +40,56 @@ export function formatSnapshot(
     const out: string[] = [];
     const symbolTable = getSymbolTable(doc);
 
+    const formatOptionsPrefix = '# format-options:';
+    const formatOptions = {
+        showDocs: false,
+        showRanges: false,
+    };
+
+    for (let line of input.lines) {
+        if (!line.startsWith(formatOptionsPrefix)) {
+            continue;
+        }
+
+        const options = line.slice(formatOptionsPrefix.length).trim().split(',');
+
+        for (let option of options) {
+            const optionName = option.trim();
+
+            if (!(optionName in formatOptions)) {
+                throw new Error(`Invalid format option: ${optionName}`);
+            }
+
+            formatOptions[optionName as keyof typeof formatOptions] = true;
+        }
+
+        break;
+    }
+
```

Comment:
This function is already getting quite long, it would be easier to read if this logic were hoisted out into a separate function that returns a `{ showDocs: boolean, showRanges: boolean}` object.